File: src/main/scala/com/mattchowning_scala/file_read_write/server/TokenGenerator.scala
package com.mattchowning_scala.file_read_write.server

import java.security.SecureRandom

object TokenGenerator {
  private val TOKEN_CHARS: String = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
  private val TOKEN_LENGTH: Int = 20
  private val RND: SecureRandom = new SecureRandom

  def generateNew: String = {
    val sb: StringBuilder = new StringBuilder(TOKEN_LENGTH)
    for (_ <- 0 until TOKEN_LENGTH) {
      val newChar = TOKEN_CHARS.charAt(RND.nextInt(TOKEN_CHARS.length()))
      sb.append(newChar)
    }
    sb.toString
  }
}

File: src/main/java/com/mattchowning/file_read_write/server/TokenGenerator.java
package com.mattchowning.file_read_write.server;

import java.security.SecureRandom;

public class TokenGenerator {

    private static final String AB = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
    private static final int TOKEN_LENGTH = 20;
    private static final SecureRandom RND = new SecureRandom();

    public static String generateNew(){
        StringBuilder sb = new StringBuilder(TOKEN_LENGTH);
        for (int i = 0; i < TOKEN_LENGTH; i++) {
            char newChar = AB.charAt(RND.nextInt(AB.length()));
            sb.append(newChar);
        }
        return sb.toString();
    }
}
